Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -3188,11 +3188,11 @@ void NetworkProcess::restoreLocalStorage(PAL::SessionID sessionID, HashMap<WebCo
     session->protectedStorageManager()->restoreLocalStorage(WTFMove(localStorageMap), WTFMove(completionHandler));
 }
 
-void NetworkProcess::fetchSessionStorage(PAL::SessionID sessionID, WebPageProxyIdentifier pageID, CompletionHandler<void(HashMap<WebCore::ClientOrigin, HashMap<String, String>>&&)>&& completionHandler)
+void NetworkProcess::fetchSessionStorage(PAL::SessionID sessionID, WebPageProxyIdentifier pageID, CompletionHandler<void(std::optional<HashMap<WebCore::ClientOrigin, HashMap<String, String>>>&&)>&& completionHandler)
 {
     CheckedPtr session = networkSession(sessionID);
     if (!session) {
-        completionHandler({ });
+        completionHandler(HashMap<WebCore::ClientOrigin, HashMap<String, String>>());
```

Comment:
... but this returns an empty HashMap.

The 2 seems like the same case, just one for local storage and the other for session storage. The lack of consistency here is fishy.

Reply:
Yeah was intentional but I'm not 100% sure about this. Since session storage is not stored on disk, it seemed that having no network process just means there's no storage so we can return an empty map. But for local storage since it is stored on disk, having no network process means we don't know if there is storage or not and we can't access it--so we should return an error (returning `std::nulltopt` will cause an error to be thrown).

What do we think about that?

Reply:
Fixed!